src/EditorFeatures/Core.Wpf/InheritanceMargin/MarginGlyph/InheritanceMargin.xaml.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Windows;
+using System.Windows.Controls;
+using Microsoft.CodeAnalysis.Editor.GoToDefinition;
+using Microsoft.CodeAnalysis.Editor.Host;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.InheritanceMargin.MarginGlyph
+{
+    internal partial class InheritanceMargin
+    {
+        private readonly IThreadingContext _threadingContext;
+        private readonly IStreamingFindUsagesPresenter _streamingFindUsagesPresenter;
+        private readonly IWaitIndicator _waitIndicator;
+        private readonly Solution _solution;
+
+        /// <summary>
+        /// Note: This name is used in xaml file.
+        /// </summary>
+        private const string SingleMemberContextMenuStyle = nameof(SingleMemberContextMenuStyle);
+
+        /// <summary>
+        /// Note: This name is used in xaml file.
+        /// </summary>
+        private const string MultipleMembersContextMenuStyle = nameof(MultipleMembersContextMenuStyle);
+
+        public static InheritanceMargin CreateForSingleMember(
+            IThreadingContext threadingContext,
+            IStreamingFindUsagesPresenter streamingFindUsagesPresenter,
+            IWaitIndicator waitIndicator,
+            Solution solution,
+            SingleMemberMarginViewModel viewModel)
```

Comment:
i'm a little confused by this signature...   why are we creating a margin for a member?  i would expect we would just have the margin, and we would create members in it.

Reply:
: ) I have some useful pictures this time,
There are two cases here,
1. If there is only one member on the line:
![image](https://user-images.githubusercontent.com/24360909/112599224-e6cd6f80-8dcc-11eb-868f-52bbd256d7b9.png)

As you can see, it directly shows the targets.

2. If there are mulitple members on the line:
![image](https://user-images.githubusercontent.com/24360909/112599326-06649800-8dcd-11eb-9ddc-e9af63f09d5e.png)
So first it pops up a menu to choose the member, and there is second level menu to select targets

Reply:
MenuItem can be nested, so there's not much reason to distinguish between single and multi. I agree with @CyrusNajmabadi that we could simplify this by just adding MenuItem to the Menu in the margin